documents/extension/funding/service.go

```diff
@@ -180,6 +194,70 @@ func (s service) DeriveFromPayload(ctx context.Context, req *clientfundingpb.Fun
 	return model, nil
 }
 
+func (s service) deleteFunding(model documents.Model, idx string) (documents.Model, error) {
+	var data Data
+	types := reflect.TypeOf(data)
+	for i := 0; i < types.NumField(); i++ {
+		jsonKey := types.Field(i).Tag.Get("json")
+		key, err := documents.AttrKeyFromLabel(labelFromJSONTag(idx, jsonKey))
+		if err != nil {
+			continue
+		}
+
+		if model.AttributeExists(key) {
+			err := model.DeleteAttribute(key)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+	}
+	return model, nil
+}
+
+// DeriveFromUpdatePayload derives Funding from clientUpdatePayload
+func (s service) DeriveFromUpdatePayload(ctx context.Context, req *clientfundingpb.FundingUpdatePayload, identifier []byte) (documents.Model, error) {
+	var fd Data
+	fd.initFundingFromData(req.Data)
+
+	model, err := s.GetCurrentVersion(ctx, identifier)
+	if err != nil {
+		apiLog.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
+
+	fd.FundingId = req.FundingId
+	idx, err := s.findFunding(model, fd.FundingId)
+	if err != nil {
+		return nil, err
+	}
+
+	// overwriting is not enough because it is not required that
+	// the funding payload contains all funding attributes
+	model, err = s.deleteFunding(model, idx)
```

Comment:
I think it is a bit cleaner to remove all current funding fields instead of only removing fields which are empty in the new version but existed in the old version.